network-testing/src/main/java/com/stripe/android/networktesting/NetworkDispatcher.kt

```diff
@@ -58,7 +59,10 @@ internal class NetworkDispatcher : Dispatcher() {
             return capturedEntry.responseFactory(testRequest)
         }
 
-        throw RequestNotFoundException("$request not mocked\n${testRequest.bodyText}")
+        val exception = RequestNotFoundException("$request not mocked\n${testRequest.bodyText}")
+        Log.d("NetworkDispatcher", "Request not found.", exception)
+        android.os.Process.killProcess(android.os.Process.myPid())
```

Comment:
Why do we need this?

Reply:
We catch exceptions where we make requests (especially in certain cases where we want to silently ignore all failures), see my other comment on the PR here: https://github.com/stripe/stripe-android/pull/7201#discussion_r1305936696

This just make sure that if we see a request to a host we're tracking, it'll really fail the test.

Most of the time it did before, but in cases like the link consumer lookup, and analytics requests, we silently ignore failures, causing the test to still pass, even though we couldn't find a request to match what's expected.

Reply:
What makes the test ignore failures? Can we update some code so that failures are not ignored, rather than forcing kill process?

Reply:
The tests aren't ignoring failures, the place that's making the network request (example: https://github.com/stripe/stripe-android/blob/262885d0c02a54430ae8f0c3966de694abc6bb0a/stripe-core/src/main/java/com/stripe/android/core/networking/DefaultAnalyticsRequestExecutor.kt#L47) is catching all failures, causing the rest of the test to continue on without failing (as you'd expect, since we don't rely on data returned from the analytics request to continue using the app).

Reply:
Ah gotcha, thanks for the explanation! Can we add a comment here which explains this all then?

Reply:
For sure. Good idea. Thanks!